pkg/testing/fixture_install.go

```diff
@@ -193,98 +206,127 @@ func (f *Fixture) installNoPkgManager(ctx context.Context, installOpts *InstallO
 	f.setClient(c)
 
 	f.t.Cleanup(func() {
-		if f.t.Failed() {
-			procs := getProcesses(f.t, `.*`)
-			dir, err := findProjectRoot(f.caller)
-			if err != nil {
-				f.t.Logf("failed to dump process; failed to find project root: %s", err)
-				return
-			}
+		f.t.Logf("[test %s] Inside fixture cleanup function", f.t.Name())
 
-			// Sub-test names are separated by "/" characters which are not valid filenames on Linux.
-			sanitizedTestName := strings.ReplaceAll(f.t.Name(), "/", "-")
+		f.cleanUpCollectDiagnostics()
 
-			filePath := filepath.Join(dir, "build", "diagnostics", fmt.Sprintf("TEST-%s-%s-%s-ProcessDump.json", sanitizedTestName, f.operatingSystem, f.architecture))
-			f.t.Logf("Dumping running processes in %s", filePath)
-			file, err := os.OpenFile(filePath, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0o644)
-			if err != nil {
-				f.t.Logf("failed to dump process; failed to create output file %s root: %s", file.Name(), err)
-				return
-			}
-			defer func(file *os.File) {
-				err := file.Close()
-				if err != nil {
-					f.t.Logf("error closing file %s: %s", file.Name(), err)
-				}
-			}(file)
-			err = json.NewEncoder(file).Encode(procs)
-			if err != nil {
-				f.t.Logf("error serializing processes: %s", err)
-			}
+		if f.cleanUpKeepInstalled() {
```

Comment:
This returns a boolean which is not documented, I cannot understand from the function name what this boolean is.